packages/blockchain-wallet-v4-frontend/src/scenes/Login/AccountUpgrade/3_3_1_GoogleAuthSetup/GoogleAuthSetup.tsx

```diff
@@ -22,10 +65,34 @@ const GoogleAuthSetup = (props) => {
           lineHeight='1.5'
         >
           <FormattedMessage
-            id='scenes.login.upgrade.upsell.header'
-            defaultMessage='GOOGLE_AUTH_SETUP'
+            id='scenes.login.upgrade.googleAuth.header'
+            defaultMessage='Set Up 2FA'
           />
         </CenteredTitle>
+        <CenteredTitle size='14px' weight={500} lineHeight='20px' color='grey900'>
+          <FormattedMessage
+            id='scenes.login.upgrade.googleAuth.text'
+            defaultMessage='With your Google Authenticator app, scan the QR code below to make a secure connection.'
+          />
+          <QrWrapper>Placeholder</QrWrapper>
+          <ClipboardWrapper>
+            {/* <CopyClipboard address={PLACEHOLDER} active={uiState.notificationActive} /> */}
+          </ClipboardWrapper>
+          <BadgesWrapper>
+            <Image name='badge-ios-appstore' width='160px' />
+            <Image name='badge-android-playstore' width='160px' />
+          </BadgesWrapper>
+        </CenteredTitle>
+
+        <ButtonNext
+          nature='primary'
+          data-e2e='nextButton'
+          fullwidth
+          height='48px'
+          onClick={() => {}}
```

Comment:
You forgot to add the functionality or is WIP?

Reply:
is wip, this doesn't have the logic yet.